### PR TITLE
Added workflow to auto-publish storybook for PRs

### PR DESCRIPTION
**Pull request by patcon**
Addresses: https://github.com/GCTC-NTGC/gc-digital-talent/issues/1691

Submitting so it's easier to review :)

### CODE REVIEW

File: .github/workflows/storybook.yml
Comment (patcon) [inline .github/workflows/storybook.yml:41]: Recommend replacing with a project token generated by someone with access to repo, though no required.

Left as-is, should still publish storybook and print its link in CI logs, just won't be able to update the PR commit status with a convenient "Details" link. Link format for this project token is always `https://<branch>--61de6c057fec2d003a9bc22f.chromatic.com` e.g. https://publish-storybook--61de6c057fec2d003a9bc22f.chromatic.com

**Review by tristan-orourke — COMMENTED** [review]
Hey, thanks Pat, this is great!

The only thing I'm not sure about is why the change of location for the generated schema file? We've been trying to avoid checking in generated files to git, because its easy to commit an out-of-date version, or forget to update one. But I imagine checking in this one speeds up the github task a lot?

**Comment by patcon** [comment]
Thanks for reviewing both!

> If we check it in, why the change of location? If we keep the new location, we should update api/README slightly.

That was maybe a bit pre-emptive, and it can be reverted :) There were a few reasons, but I could get feedback on that in another PR. Sound ok?

**Comment by patcon** [comment]
Ok, made those changes back to original path, and should be good for merge if it suits y'all. (Updated the projectToken in one other place as well)

**Comment by tristan-orourke** [comment]
I think I can see where you're going with the moving of the schema, but I like the idea of handling that in a separate PR. Thanks!

File: admin/.storybook/main.js
Comment (yonikid15) [inline admin/.storybook/main.js:22]: Does the `MERGE_STORYBOOKS` need to get added to the admin/.env file? If that's the case we should add it to the .env/example file.

**Review by yonikid15 — APPROVED** [review]
Wow this is awesome! Thanks for sharing this 👍

**Comment by yonikid15** [comment]
Woops accidentally hit approve! Just had one question.

File: admin/.storybook/main.js
Comment (yonikid15) [reply admin/.storybook/main.js:22]: Nvm, just found out that it defaults to false because its not in our .env files, and its only the github action workflow that sets this variable.

Comment (patcon) [reply admin/.storybook/main.js:22]: I actually think it's a great idea to keep it in a `admin/.env.example`! I like the idea of one file that's am authoritative map for all the envvars the code uses :)

**Comment by patcon** [comment]
Sorry, made that change, so needs another approval. Hopefully after this PR gets merged, GitHub won't require approval for every CI run...!

---

I see that this "needing approval" thing is something GitHub changed recently to deal with people running crypto-miners in PRs. Since it's not a security concern, can we ratchet it down a notch for repo-wide (or org-wide?) settings, so

> - [x] **Require approval for first-time contributors who are new to GitHub.**
> - [ ] Require approval for first-time contributors.

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#configuring-required-approval-for-workflows-from-public-forks

**Comment by tristan-orourke** [comment]
The GitHub action approval settings are currently set to require approval for First-time contributors, which I think is reasonable. Hopefully after we accept this PR, it won't block you any more.

**Review by tristan-orourke — APPROVED** [review]
Yeah, looks good to me, and we all love the feature.

**Pull request merged**
